shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -163,17 +162,13 @@ private void updateEditingState() {
         currentValue.composingStart,
         currentValue.composingEnd);
 
-    mRepeatCheckNeeded = true;
-    mLastSentTextEditngValue = currentValue;
+    mLastKnownTextEditingValue = currentValue;
   }
 
-  // This should be called whenever a change could have been made to
-  // the value of mEditable, which will make any call of updateEditingState()
-  // ineligible for repeat checking as we do not want to skip sending real changes
-  // to the framework.
-  public void markDirty() {
-    // Disable updateEditngState's repeat-update check
-    mRepeatCheckNeeded = false;
+  // Called when the current text editing state held by the text input plugin is overwritten by a
+  // newly received value from the framework.
+  public void didUpdateEditingValue() {
```

Comment:
The android text input plugin doesn't seem to completely overwrite the input state with the newly received value, the composing range is ignored. If the text is composing and the user replaces the entire string with something else, would that put the input method in a broken state?

Reply:
Have you tried it out?  Ignoring the composing region definitely seems incorrect.  If there's no visible bug, maybe it's fixing itself when subsequent edits happen.

Reply:
I tried a 3rd party japanese IME that uses composing region, it's not working in flutter apps (not composing at all). I'll see to there's an easy fix.